discard_test.go

```diff
@@ -46,4 +47,12 @@ func TestComparable(t *testing.T) {
 	if a != b {
 		t.Fatal("any two discardLogSink must be equal")
 	}
+
+	if b := Discard().WithContextValues(ContextKey{Key: 1, Name: "one"}); b != a {
```

Comment:
This seems like it should read as `a.WithContextValues(...`, no?   same below.

Reply:
Ultimately it's the same, but you are right, starting with `a` is a bit closer to the intent. Will change.